subsys/bluetooth/audio/bass.c

```diff
@@ -0,0 +1,1373 @@
+		}
+	}
+
+	if (bass_cbs != NULL && bass_cbs->biginfo != NULL) {
+		bass_cbs->biginfo(&state->state, biginfo);
+	}
+}
+
+static struct bt_le_per_adv_sync_cb pa_sync_cb =  {
+	.synced = pa_synced,
+	.term = pa_terminated,
+	.recv = pa_recv,
+	.biginfo = biginfo_recv
+};
+
+static void pa_timer_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct bass_recv_state_internal *recv_state = CONTAINER_OF(
+		dwork, struct bass_recv_state_internal, pa_timer);
+
+	BT_DBG("PA timeout");
+
+	__ASSERT(recv_state, "NULL receive state");
+
+	if (recv_state->state.pa_sync_state == BT_BASS_PA_STATE_INFO_REQ) {
+		recv_state->state.pa_sync_state = BT_BASS_PA_STATE_NO_PAST;
+	} else {
+		int err = bt_le_per_adv_sync_delete(recv_state->pa_sync);
+
+		if (err != 0) {
+			BT_ERR("Could not delete BASS pa_sync");
+		}
+
+		recv_state->state.pa_sync_state = BT_BASS_PA_STATE_FAILED;
+
+	}
+
+	bt_debug_dump_recv_state(recv_state);
+	net_buf_put_recv_state(recv_state);
+	bass_notify_receive_state(recv_state);
+}
+
+static int bis_sync(struct bass_recv_state_internal *state)
+{
+	int err;
+	struct bt_iso_big_sync_param param;
+	struct bt_iso_chan *bis_channels[1] = { &bis_iso_chan };
+
+	if (state->big != NULL) {
+		return -EALREADY;
+	}
+
+	bis_iso_qos.tx = NULL;
+
+	param.bis_channels = bis_channels;
+	param.num_bis = ARRAY_SIZE(bis_channels);
+	param.encryption = false;
+	param.bis_bitfield = aggregated_bis_syncs_get(state);
+	param.mse = 0;
+	param.sync_timeout = interval_to_sync_timeout(state->iso_interval);
+
+	BT_DBG("Bitfield %x", param.bis_bitfield);
+
+	if (param.bis_bitfield == 0) {
+		/* Don't attempt to sync anything */
+		return 0;
+	} else if (param.bis_bitfield == BT_BASS_BIS_SYNC_NO_PREF) {
+		param.bis_bitfield = 0;
+		/* Attempt to sync to all BISes */
+		for (int i = 0; i < state->biginfo_num_bis; i++) {
+			param.bis_bitfield |= BIT(i);
+		}
+	} else {
+		/* Ensure that we don't attempt to sync to more BISes that is possible */
+		uint32_t max_bis_bitfield = 0;
+
+		for (int i = 0; i < state->biginfo_num_bis; i++) {
+			max_bis_bitfield |= BIT(i);
+		}
+
+		param.bis_bitfield &= max_bis_bitfield;
+	}
+
+	/* TODO: For now we only support syncing to a single BIS,
+	 * so only sync to first BIS
+	 */
+	for (int i = 0; i < state->biginfo_num_bis; i++) {
+		if (param.bis_bitfield & BIT(i)) {
+			param.bis_bitfield = BIT(i);
+			break;
+		}
+	}
+
+	err = bt_iso_big_sync(state->pa_sync, &param, &state->big);
+	if (err != 0) {
+		return err;
+	}
+
+	/* We could start a timer for BIG sync but there is no way to let the
+	 * client know if it has timed out, so it doesn't really matter.
+	 */
+	return 0;
+}
+
+static int bis_sync_cancel(struct bass_recv_state_internal *state)
+{
+	int err;
+
+	if (!state->big) {
+		return 0;
+	}
+
+	err = bt_iso_big_terminate(state->big);
+	if (err != 0) {
+		return err;
+	}
+
+	state->big = NULL;
+
+	return 0;
+}
+
+static void bass_pa_sync_past(struct bt_conn *conn,
```

Comment:
Eh, not really. 

AFAIK we don't really have a naming scheme for static functions, although it could be argued that 

`bt_xxxx_yyyy` is public or internal (e.g. `bt_bass_yyyy`)
`xxxx_yyyy` is static (e.g. `bass_yyyy`)

but looking at the general codebase we also often just do `yyyy`. 

I'll add it to the BT subsys discussion.